src/browser-attributes-span-processor.ts

```diff
@@ -1,13 +1,19 @@
 import { Span } from '@opentelemetry/api';
 import { SpanProcessor } from '@opentelemetry/sdk-trace-base';
 
+export interface BrowserAttributesSpanProcessorConfig {
+  getRoute?: () => string;
+}
 /**
  * A {@link SpanProcessor} that adds browser specific attributes to each span
  * that might change over the course of a session.
  * Static attributes (e.g. User Agent) are added to the Resource.
  */
 export class BrowserAttributesSpanProcessor implements SpanProcessor {
-  constructor() {}
+  getRoute: () => string;
```

Comment:
I was thinking about this a little bit and I feel like this in itself would make a great span processor that we can export from the distro, it would be consumed by passing this processor to the `spanProcessor` config option itself. This way we keep the surface area of the API more consistent and get this ready for reuse / contribute it upstream to OTel.

We should extend our `spanProcessor` option to be `spanProcessors` that can take multiple span processors but we can address that in a separate PR.

Reply:
That sounds very sensible to me, how would you like me to action in the context of this PR?

Reply:
Would you be up for reworking this PR by creating a new span processor for this and showing an example of how to use it in the example app? 

I will open up an issue to accept multiple span processors and open up a separate PR myself for that.

Reply:
Happy to spend some time on that, one question: should I remove the `page.route` attribute from this processor?